instrumentation/jvm-metrics/src/main/java/com/splunk/opentelemetry/instrumentation/jvmmetrics/AllocatedMemoryMetrics.java

```diff
@@ -53,13 +52,22 @@ private static boolean hasComSunThreadMXBean() {
     }
   }
 
+  private static boolean mxBeanTypeIsCompatible() {
+    try {
+      Class<?> mxBeanClass = Class.forName("com.sun.management.ThreadMXBean");
+      return mxBeanClass.isInstance(ManagementFactory.getThreadMXBean());
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
   private static boolean isThreadAllocatedMemoryEnabled() {
-    ThreadMXBean threadBean = (ThreadMXBean) ManagementFactory.getThreadMXBean();
+    ThreadMXBean threadBean = (com.sun.management.ThreadMXBean) ManagementFactory.getThreadMXBean();
```

Comment:
unnecessary qualified class name in the cast

Reply:
This was intentional. This code is already rather hard to follow because there are two classes called `ThreadMXBean` in different packages, and I thought that the fully qualified cast actually helped here.